src/main/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/DeleteRecommendationService.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.hmpps.makerecalldecisionapi.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.domain.makerecalldecisions.DeleteRecommendationResponse
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.exception.NoDeletedRecommendationRationaleException
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.jpa.repository.RecommendationRepository
+import uk.gov.justice.digital.hmpps.makerecalldecisionapi.jpa.repository.RecommendationStatusRepository
+import java.util.Collections
+
+@Transactional
+@Service
+internal class DeleteRecommendationService(
+  val recommendationRepository: RecommendationRepository,
+  val recommendationStatusRepository: RecommendationStatusRepository,
+  @Value("\${mrd.url}") private val mrdUrl: String? = null,
+) {
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
+  suspend fun getDeleteRecommendationResponse(
+    crn: String,
+  ): ResponseEntity<DeleteRecommendationResponse> {
+    val recommendations = recommendationRepository.findByCrn(crn)
+    Collections.sort(recommendations)
```

Comment:
You're probably getting a code analysis warning here. Better to do:
`val recommendations = recommendationRepository.findByCrn(crn).sorted()`